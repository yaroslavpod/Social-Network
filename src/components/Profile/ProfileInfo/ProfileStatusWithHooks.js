import classes from "./ProfileStatus.module.css"
import * as React from "react";
import {useState,useEffect} from "react";


const ProfileStatusWithHooks = ({updateUserStatus,...props}) => {

    const [editMode,setEditMode] = useState(false);
    const [status,setStatus] = useState(props.status);

    const activatedEditMode =()=>{
        setEditMode(true)
    }
    const deactivatedEditMode =()=>{
        setEditMode(false);
        updateUserStatus(status);
    }
    const handleFocus = (e) => {
        e.currentTarget.select();
    }
    const onStatusChange = (e) => {
        setStatus(e.currentTarget.value)
    }
    useEffect(()=>{
        if(status !== props.status){
            setStatus(props.status)
        }
    },[props.status])



    return (
        <div className={classes.status}>
            {!editMode &&
            <div>
                <span onClick={activatedEditMode}>{props.status || "---"}</span>
            </div>
            }
            {editMode  &&
            <div>
                <input  onChange={onStatusChange} value={status}  onBlur={deactivatedEditMode} autoFocus={true} onFocus={handleFocus} />

            </div>
            }

        </div>)

}

export default ProfileStatusWithHooks