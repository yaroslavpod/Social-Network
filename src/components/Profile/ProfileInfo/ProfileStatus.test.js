import React from "react";
import {create} from "react-test-renderer";
import {act} from "@testing-library/react";
import ProfileStatusWithHooks from "./ProfileStatusWithHooks";

describe("Profile status component", () => {

    test("it shows the expected text when clicked", () => {
        let component;
        act(()=>{
            component = create(<ProfileStatusWithHooks status = "Hell0"/>)
        })
        const instance = component.root;
        let status =instance._fiber.pendingProps.status;
        expect(status).toBe("Hell0");
    });
    test("span shuold be visible", () => {
        let component;
        act(()=>{
            component = create(<ProfileStatusWithHooks status = "Hell0"/>)
        })
        const instance = component.root;
        const span = instance.findByType("span")
        expect(span).not.toBeNull();
    });
    test("after creation input shouldn't be visible", () => {
        let component;
        act(()=>{
            component = create(<ProfileStatusWithHooks status = "Hell0"/>)
        })
        const instance = component.root;
        expect(()=>{
            const input = instance.findByType("input")
        }).toThrow();
    });
    test("Status should be from props", () => {
        let component;
        act(()=>{
            component = create(<ProfileStatusWithHooks status = "Hell0"/>)
        })
        const instance = component.root;
        const span = instance.findByType("span")
        expect(span.children[0]).toBe("Hell0");
    });
    test("In edit mode should be displayed input instead of span", () => {
        let component;
        act(()=>{
            component = create(<ProfileStatusWithHooks status = "Hell0"/>)
        })
        const instance = component.root;
        const span = instance.findByType("span");
        act(()=>{
            span.props.onClick()
        })
        const input = instance.findByType("input")
        expect(input.props.value).toBe("Hell0");
    });
    test("callback should be called", () => {
        let component;
        const mockCallback = jest.fn();
        act(()=>{
            component = create(<ProfileStatusWithHooks status = "Hell0" updateUserStatus ={mockCallback}/>)
        })
        const instance = component.root;
        const span = instance.findByType("span");
        act(()=>{
            span.props.onClick()
        })
        const input = instance.findByType("input")
        input.props.onBlur();
        expect(mockCallback.mock.calls.length).toBe(1);
    });

});