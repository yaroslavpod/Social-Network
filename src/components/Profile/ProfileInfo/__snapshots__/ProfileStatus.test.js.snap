// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile status component it shows the expected text when clicked 1`] = `
<div
  className="status"
>
  <div>
    <span
      onClick={[Function]}
    >
      Hell0
    </span>
  </div>
</div>
`;
